onmt/train_single.py

```diff
@@ -122,12 +188,15 @@ def _train_iter():
             while True:
                 batch = batch_queue.get()
                 semaphore.release()
+                # Maybe move batch to specified device
+                batch_to(batch, device_id)
```

Comment:
Hmm, so this is done quite in the opposite way as before, right?
Before: the batch_producer process would send directly to the gpus
Now: all gpu queues are originally on cpu, and then moved when retrieved, right?

Reply:
Yes.

Reply:
Ok, so that would basically solve Waino's "exclusive mode" issue, right?

Reply:
I'm afraid not? I just thought by putting data originally on cpu, then moved to correspond gpu when retrieved will decrease the usage needs of gpu memory as we need a large Queue size to prepare enough batches. And the previous method creates data on default gpu but always needs to send them to corresponding indexed gpu, therefore the inter-process data transfer is needed anyway.

Reply:
We'll need to check this more carefully, but I think it does, because we will have only a single process per GPU in such a setup, hence compatible with GPU exclusive mode.